spock-core/src/main/java/org/spockframework/runtime/extension/builtin/TimeoutInterceptor.java

```diff
@@ -123,4 +130,94 @@ public void run() {
       throw saved;
     }
   }
+
+  private void logUnsuccessfulInterrupt(String methodName, long now, long timeoutAt, long waitMillis, int unsuccessfulAttempts) {
+    System.err.printf(
+      "[spock.lang.Timeout] Method '%s' has not stopped after timing out %1.2f seconds ago - interrupting. Next try in %1.2f seconds.\n%n",
+      methodName,
+      Duration.ofNanos(now - timeoutAt).toMillis() / 1000.,
+      waitMillis / 1000.
+    );
+
+    if (unsuccessfulAttempts <= configuration.maxInterruptAttemptsWithThreadDumps) {
+      logThreadDumpOfCurrentJvm();
+      configuration.interruptAttemptListeners.forEach(Runnable::run);
+
+      if (unsuccessfulAttempts == configuration.maxInterruptAttemptsWithThreadDumps) {
+        System.out.println("[spock.lang.Timeout] No further thread dumps will be logged and no timeout listeners will be run, as the number of unsuccessful interrupt attempts exceeds configured maximum of logged attempts");
+      }
+    }
+  }
+
+  private void logMethodTimeout(String methodName, double timeoutSeconds) {
+    System.err.printf(
+      "[spock.lang.Timeout] Method '%s' timed out after %1.2f seconds.%n",
+      methodName,
+      timeoutSeconds
+    );
+
+    configuration.interruptAttemptListeners.forEach(Runnable::run);
+  }
+
+  private void logThreadDumpOfCurrentJvm() {
+    if (configuration.printThreadDumpsOnInterruptAttempts) {
+      StringBuilder sb = new StringBuilder();
+      try {
+        threadDumpCollector.appendThreadDumpOfCurrentJvm(sb);
+        System.err.println(removeThisThread(sb.toString()));
+      } catch (Throwable e) {
```

Comment:
I think it would be best to add our version of `org.junit.platform.commons.util.UnrecoverableExceptions#rethrowIfUnrecoverable` to `org.spockframework.util.ExceptionUtil` and use it here.

Reply:
Done in https://github.com/spockframework/spock/pull/1658/commits/859509b1a1a3bbfa856f58f63e0926fb4e5e4318